pkg/helpers/tokencmd/basicauth.go

```diff
@@ -54,13 +71,17 @@ func (c *BasicChallengeHandler) HandleChallenge(requestURL string, headers http.
 	missingUsername := len(username) == 0
 	missingPassword := len(password) == 0
 
+	isBasic, realm := basicRealm(headers)
+	if missingUsername && isBasic {
```

Comment:
you already know basic auth is possible at this point so you can move getting the realm to the point of use and fail if username is missing

Reply:
right, updated.